docs/en/development/data_structures.md

```diff
@@ -5,4 +5,879 @@ slug: /internals
+:::
+
+#### 3.1.4 SessionInfo
+
+Records specific metadata about the client session so that it can be viewed with the `juicefs status` command. This is specified as
+
+```go
+type SessionInfo struct {
+	Version    string // JuiceFS version
+	HostName   string // Host name
+	MountPoint string // path to mount point. S3 gateway and WebDAV server are "s3gateway" and "webdav" respectively
+	ProcessID  int    // Process ID
+}
+```
+
+This structure is serialized into JSON format and stored in the metadata engine.
+
+#### 3.1.5 Node
+
+Records information about the attributes of each file, as follows
+
+```go
+type Attr struct {
+	Flags     uint8  // reserved flags
+	Typ       uint8  // type of a node
+	Mode      uint16 // permission mode
+	Uid       uint32 // owner id
+	Gid       uint32 // group id of owner
+	Rdev      uint32 // device number
+	Atime     int64  // last access time
+	Mtime     int64  // last modified time
+	Ctime     int64  // last change time for meta
+	Atimensec uint32 // nanosecond part of atime
+	Mtimensec uint32 // nanosecond part of mtime
+	Ctimensec uint32 // nanosecond part of ctime
+	Nlink     uint32 // number of links (sub-directories or hardlinks)
+	Length    uint64 // length of regular file
+
+	Parent    Ino  // inode of parent; 0 means tracked by parentKey (for hardlinks)
+	Full      bool // the attributes are completed or not
+	KeepCache bool // whether to keep the cached page or not
+}
+```
+
+A few of the fields that need clarification.
+
+- Atime/Atimensec: set only when the file is created and when `SetAttr` is actively called, while accessing and modifying the file normally does not affect the Atime value
+- Nlink
+  - Directory file: initial value is 2 ('.' and '..'), add 1 for each subdirectory
+  - Other files: initial value is 1, add 1 for each hard link created
+- Length
+  - Directory file: fixed at 4096
+  - Soft link (symbolic link) file: length of the string of the path to which the link points
+  - Other files: the length of the actual content of the file
+
+This structure is usually encoded in binary format and stored in the metadata engine.
+
+#### 3.1.6 Edges
+
+Records information about each edge in the file tree, as follows
+
+```
+parentInode, name -> type, inode
+```
+
+where parentInode is the inode number of the parent directory, and the others are the name, type, and inode number of the child files, respectively.
+
+#### 3.1.7 LinkParent
+
+Records the parent directory of some files. The parent directory of most files is recorded in the Parent field of the attribute; however, for files that have been created with hard links, there may be more than one parent directory, so the Parent field is set to 0, and all parent inodes are recorded independently, as follows
+
+```
+inode -> parentInode, links
+```
+
+where links is the count of the parentInode, because multiple hard links can be created in the same directory, and these hard links share the inode.
```

Comment:
```suggestion
where links is the count of the parentInode, because multiple hard links can be created in the same directory, and these hard links share one inode.
```